core/src/main/java/com/novoda/noplayer/player/PlayerFactory.java

```diff
@@ -18,78 +19,75 @@
+        this.exoPlayerCreator = exoPlayerCreator;
+        this.mediaPlayerCreator = mediaPlayerCreator;
     }
 
     public Player create() {
-        return create(DrmHandler.NO_DRM);
+        return create(DrmType.NONE, DrmHandler.NO_DRM);
     }
 
-    public Player create(DrmHandler drm) {
-        for (PlayerType playerType : prioritisedPlayers) {
-            if (playerType.supports(drm)) {
-                return createPlayerForType(playerType, drm);
+    public Player create(DrmType drmType, DrmHandler drmHandler) {
+        for (PlayerType player : prioritizedPlayerTypes) {
+            if (player.supports(drmType)) {
+                return createPlayerForType(player, drmType, drmHandler);
             }
         }
-        throw UnableToCreatePlayerException.contentNotSupported();
+        throw UnableToCreatePlayerException.unknownDrmType(drmType);
     }
 
-    private Player createPlayerForType(PlayerType playerType, DrmHandler drm) {
+    private Player createPlayerForType(PlayerType playerType, DrmType drmType, DrmHandler drmHandler) {
         switch (playerType) {
             case MEDIA_PLAYER:
-                return createMediaPlayer();
+                return mediaPlayerCreator.createMediaPlayer(context);
             case EXO_PLAYER:
-                return createExoPlayer(createDrmSessionCreatorFor(drm));
+                DrmSessionCreator drmSessionCreator = createDrmSessionCreatorFor(drmType, drmHandler);
+                return exoPlayerCreator.createExoPlayer(context, drmSessionCreator);
             default:
                 throw UnableToCreatePlayerException.unknownPlayerType(playerType);
         }
     }
 
-    private DrmSessionCreator createDrmSessionCreatorFor(DrmHandler drm) {
-        if (drm == DrmHandler.NO_DRM) {
-            return new NoDrmSessionCreator();
-        } else if (drm instanceof StreamingModularDrm) {
-            ProvisionExecutor provisionExecutor = ProvisionExecutor.newInstance();
-            ProvisioningModularDrmCallback mediaDrmCallback = new ProvisioningModularDrmCallback((StreamingModularDrm) drm, provisionExecutor);
-            return new StreamingDrmSessionCreator(mediaDrmCallback);
-        } else if (drm instanceof DownloadedModularDrm) {
-            return new LocalDrmSessionCreator((DownloadedModularDrm) drm);
-        } else {
-            throw UnableToCreatePlayerException.unknownDrmHandler(drm);
+    private DrmSessionCreator createDrmSessionCreatorFor(DrmType drmType, DrmHandler drmHandler) {
+        switch (drmType) {
+            case NONE:
+            case WIDEVINE_CLASSIC:
+                return new NoDrmSessionCreator();
+            case WIDEVINE_MODULAR_STREAM:
+                ProvisionExecutor provisionExecutor = ProvisionExecutor.newInstance();
+                ProvisioningModularDrmCallback mediaDrmCallback = new ProvisioningModularDrmCallback((StreamingModularDrm) drmHandler, provisionExecutor);
```

Comment:
a good question. no it's not safe, but we want to remove this anyway ([existing code](https://github.com/novoda/no-player/pull/16/files#diff-64e84b7d1365da4293b3daa3870cc5a7L57))

>Discussed with @ouchadam and probably the DrmHandler will be removed from the creation of the player - that's why I only tested the "happy" paths for the creation tests (i.e. not mixing the wrong DrmHandler for the given DrmType).